.prettierignore

```diff
@@ -0,0 +1,18 @@
+/.cache
+/art
+/coverage
+/doc
+/node/lib
+/node/src/fbs
+/rust
+/target
+/worker/deps
```

Comment:
Why not ignore whole /worker ?

Reply:
Since `.prettierignore` uses `.ignignore` syntax, I'd say ignore everything and un-ignore only things that we do want to check instead, that is often a better long-term solution.

Reply:
We cannot ignore `worker/` because we want to include `worker/scripts/clang-format.mjs`. More about this:

* Unfortunately there is no way to tell prettier which explicit folders/paths it must observe/write **via .prettierrs conf file**.
* When calling `prettier` command we pass the explicit list of folders/paths via argument (see `formatNode()` in `npm-scripts.mjs`):
  ```js
  function formatNode() {
	logInfo('formatNode()');

	const paths = [
		'npm-scripts.mjs',
		'.eslintrc.js',
		'node/src',
		'worker/scripts/clang-format.mjs',
	];

	executeCmd(`prettier ${paths.join(' ')} --write`);
  }
  ```
* The problem is that we also want to make Prettier work on our text editors to automatically format code. And the text editor just runs `prettier --write` on any folder/file included in whatever root folder in which `.prettierrs` is.
* So if I open any TypeScript file in Rust `frontend-examples/` folder and edit it in my text editor, Prettier will format it, and we don't want it (at least not now since it would require that we also run ESlint on those files and that's another story).
* So the only way to limit the forlders/paths in which Prettier works when using the text editor is by blacklisting some of them via `.prettierrcignore`. That's sad but that's what we have. Here a feature request about it that was rejected due to "inactivity" (as tons of other similar ones): https://github.com/prettier/prettier/issues/8506
* In the other side, Prettier is supposed to also ignore everything in `.gitignore` according to the docs (https://prettier.io/docs/en/ignore.html) but it's not true. Probably it only works if there is no `.prettierrcignore` file.

Reply:
What I'm suggesting then is the following in `.prettierignore` (if it works the way I expect it to):
```
*
!/npm-scripts.mjs
!/.eslintrc.js
!/node/src
!/worker/scripts/clang-format.mjs
```

This will ignore everything and then strategically include just a few things we actually care about.

Reply:
I tried that syntax in `.prettierrcignore` and it didn't work for me. Will try again.

Reply:
Nope. This `.prettierrcignore` file:

```
*
!/npm-scripts.mjs
!/.eslintrc.js
!/node/src
!/worker/scripts/clang-format.mjs
```

- Doesn't ignore `/npm-scripts.mjs` (good).
- Ignores any file in `/node/src` (bad).
- Ignores `worker/scripts/clang-format.mjs` (bad).

Tested right now.

Reply:
Very strange, I hope there is a way to make it work, that would be the cleanest solution.

Reply:
This is what I have to do to make Prettier reformat `node/src/*` TS files and ignore `node/lib/*` JS compiled files:

```
/*
!/npm-scripts.mjs
!/.eslintrc.js
!/node
/node/lib
```

Terrible, and still I don't know how to make it work with `/worker/scripts/clang-format.mjs`.

Reply:
I give up. I've tested everything. It's buggy.